modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -607,7 +609,7 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
         Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
         if (assignment == null || assignment.isBucketEmpty()) {
             if (createAssignment) {
-                if (experiment.getState() == Experiment.State.PAUSED) {
+                if (experiment.getState() == Experiment.State.PAUSED || !checkUserCapForRapidExperiment(experiment)) {
```

Comment:
@abhishekjain88: just a nit: should the method name be more descriptive, as opposed to just checking: "checkUserCapForRapidExperiment(...)"? As in, "isUserCapForRapidExperimentExceeded(...)"?